src/plugins/intel_cpu/src/nodes/reduce.cpp

```diff
@@ -2029,11 +2030,27 @@ void Reduce::initSupportedPrimitiveDescriptors() {
                 dstMemoryDescs.push_back(config.outConfs[i].getMemDesc());
             }
 
-            auto factory = std::make_shared<ReduceExecutorFactory>(reduceAttrs, srcMemoryDescs, dstMemoryDescs,
+            auto attrs = reduceAttrs;
+            bool apply_ref = customImplPriorities.size() > 0 && customImplPriorities[0] == ref;
+            bool single_axis_only = one_of(algorithm, Algorithm::ReduceSum, Algorithm::ReduceMax,
+                                                      Algorithm::ReduceMin, Algorithm::ReduceProd);
+            // For the case of empty input, transformations ConvertReduceProd(Min, Max, Sum) are disabled to avoid empty output.
+            // So these 4 reduce modes with empty input reducing more than one axis are not supported by acl executor. Then
+            // factory->isEmpty() returns true, supportedPrimitiveDescriptors is emtpy. Though we don't actually need these acl
+            // kernels in execution, here we pass a fake axis {1} to pass assertion of "!supportedPrimitiveDescriptors.empty()"
+            // in Node::filterSupportedPrimitiveDescriptors().
+            if (!apply_ref && !isDynamicNode() && single_axis_only) {
+                const auto& src_shape = getInputShapeAtPort(REDUCE_DATA).getStaticDims();
+                if (shape_size(src_shape) == 0) {
+                    attrs.axes = {1};
+                }
+            }
+            auto factory = std::make_shared<ReduceExecutorFactory>(attrs, srcMemoryDescs, dstMemoryDescs,
```

Comment:
That logic looks like unnecessary complication. Cannot we just set canUseAclExecutor to false for that scenario?

Reply:
Applied. Now the logic in initSupportedPrimitiveDescriptors goes in a ref way, and test cases of ARM with empty input are modified accordingly. Thanks Dmitry!